martin/src/cog/image.rs

```diff
@@ -0,0 +1,327 @@
+    /// A `MartinResult` containing the tile data as a `Vec<u8>` (PNG bytes) or an error.
+    #[expect(clippy::cast_sign_loss, clippy::cast_possible_truncation)]
+    pub fn get_tile(
+        &self,
+        decoder: &mut Decoder<File>,
+        xyz: TileCoord,
+        nodata: Option<f64>,
+        path: &Path,
+    ) -> MartinResult<TileData> {
+        decoder
+            .seek_to_image(self.ifd_index)
+            .map_err(|e| CogError::IfdSeekFailed(e, self.ifd_index, path.to_path_buf()))?;
+
+        let tile_idx;
+        if let Some(idx) = self.get_tile_index(xyz) {
+            tile_idx = idx;
+        } else {
+            return Ok(Vec::new());
+        }
+        let decode_result = decoder.read_chunk(tile_idx).map_err(|e| {
+            CogError::ReadChunkFailed(e, tile_idx, self.ifd_index, path.to_path_buf())
+        })?;
+        let color_type = decoder
+            .colortype()
+            .map_err(|e| CogError::InvalidTiffFile(e, path.to_path_buf()))?;
+
+        let (tile_width, tile_height) = decoder.chunk_dimensions();
+        let (data_width, data_height) = decoder.chunk_data_dimensions(tile_idx);
+
+        //FIXME: do more research on the not u8 case, is this the right way to do it?
+        let png_file_bytes = match (decode_result, color_type) {
+            (DecodingResult::U8(vec), tiff::ColorType::RGB(_)) => rgb_to_png(
+                vec,
+                (tile_width, tile_height),
+                (data_width, data_height),
+                3,
+                nodata.map(|v| v as u8),
+                path,
+            ),
+            (DecodingResult::U8(vec), tiff::ColorType::RGBA(_)) => rgb_to_png(
+                vec,
+                (tile_width, tile_height),
+                (data_width, data_height),
+                4,
+                nodata.map(|v| v as u8),
+                path,
+            ),
+            (_, _) => Err(CogError::NotSupportedColorTypeAndBitDepth(
+                color_type,
+                path.to_path_buf(),
+            )),
+            //todo do others in next PRs, a lot of discussion would be needed
+        }?;
+        Ok(png_file_bytes)
+    }
+
+    pub fn ifd_index(&self) -> usize {
+        self.ifd_index
+    }
+
+    fn get_tile_index(&self, xyz: TileCoord) -> Option<u32> {
+        if xyz.y >= self.tiles_down || xyz.x >= self.tiles_across {
+            return None;
+        }
+
+        let tile_idx = xyz.y * self.tiles_across + xyz.x;
+        Some(tile_idx)
+    }
+}
+
+/// Converts RGB/RGBA tile data to PNG format.
+///
+/// # Arguments
+/// * `data` - Raw pixel data from TIFF decoder
+/// * `tile_width`, `tile_height` - Expected tile dimensions
+/// * `data_width`, `data_height` - Actual data dimensions
+/// * `components_count` - Number of color components (3 for RGB, 4 for RGBA)
+/// * `nodata` - Optional nodata value to make transparent
+/// * `path` - File path for error reporting
+///
+/// # Returns
+/// PNG-encoded tile data as bytes
```

Comment:
function description's first line is obvious enough, no need to repeat.  You may want to clean this for other functions as well

```suggestion
```